oak/server/oak_manager.cc

```diff
@@ -65,22 +72,26 @@ void OakManager::CreateEnclave(const std::string& node_id, const std::string& mo
   initialize_input->set_module(module);
   asylo::Status status = enclave_manager_->LoadEnclave(node_id, *enclave_loader_, config);
   if (!status.ok()) {
-    LOG(QFATAL) << "Could not load enclave " << enclave_path_ << ": " << status;
+    LOG(ERROR) << "Could not load enclave " << enclave_path_ << ": " << status;
+    return status.ToOtherStatus<grpc::Status>();
   }
   LOG(INFO) << "Enclave created";
+  return grpc::Status::OK;
 }
 
-oak::InitializeOutput OakManager::GetEnclaveOutput(const std::string& node_id) {
+grpc::Status OakManager::GetEnclaveOutput(const std::string& node_id, oak::InitializeOutput* out) {
   LOG(INFO) << "Initializing enclave";
   asylo::EnclaveClient* client = enclave_manager_->GetClient(node_id);
   asylo::EnclaveInput input;
   asylo::EnclaveOutput output;
   asylo::Status status = client->EnterAndRun(input, &output);
   if (!status.ok()) {
-    LOG(QFATAL) << "EnterAndRun failed: " << status;
+    LOG(ERROR) << "EnterAndRun failed: " << status;
+    return status.ToOtherStatus<grpc::Status>();
   }
   LOG(INFO) << "Enclave initialized";
-  return output.GetExtension(oak::initialize_output);
+  *out = output.GetExtension(oak::initialize_output);
```

Comment:
Uhmm, is this safe? Who owns this object after this method returns? And when are we rewriting all this stuff in Rust? :)

Reply:
No longer relevant with a shift to `StatusOr`, but the caller owns the object and lends it mutably to this function to fill in.